packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1496,92 +1543,271 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   }
 }
 
+/// An edge of a box, such as top, bottom, left or right, used to compute
+/// [SemanticsNode]s that overlap vertically or horizontally.
+class _BoxEdge implements Comparable<_BoxEdge> {
+  _BoxEdge({
+    @required this.isStart,
+    @required this.offset,
+    @required this.node,
+  }) : assert(isStart != null),
+       assert(offset != null),
+       assert(node != null);
+
+  /// True if the edge comes before the seconds edge along the traversal
+  /// direction, and false otherwise.
+  ///
+  /// This field is never null.
+  ///
+  /// For example, in LTR traversal the left edge's [isStart] is set to true,
+  /// the right edge's [isStart] is set to false. When considering vertical
+  /// ordering of boxes, the top edge is the start edge, and the bottom edge is
+  /// the end edge.
+  final bool isStart;
```

Comment:
I believe in other parts of the flutter framework we call this the leading edge. Maybe rename for consistency?

Reply:
Done.